core/src/main/scala/com/microsoft/azure/synapse/ml/io/binary/BinaryFileFormat.scala

```diff
@@ -228,7 +228,7 @@ class BinaryOutputWriter(val path: String,
   override def write(row: InternalRow): Unit = {
     val bytes = row.getBinary(bytesCol)
     val filename = row.getString(pathCol)
-    val nonTempPath = new Path(path).getParent.getParent.getParent.getParent.getParent
```

Comment:
The bunch of `.getParent` calls are unnecessary. Spark's output committer will move the files to the main output directory. Having these calls doesn't work in cloud environment.